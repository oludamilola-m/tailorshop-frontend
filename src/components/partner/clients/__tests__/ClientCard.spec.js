import React from "react";
import ClientCard from "../ClientCard";
import ReactDOM from "react-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import faker from "faker";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const client = {
  uuid: "uuid",
  first_name: "Olumuyiwa",
  last_name: "Margaret",
  phone_number: "0801111111",
  address: "Gilder strasse 109, 36392 Berlin",
};

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<ClientCard />, div);
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <ClientCard client={client} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("displays client properties", () => {
  const { getByTestId } = render(
    <Router>
      <ClientCard client={client} />
    </Router>
  );

  expect(getByTestId("client-name")).toHaveTextContent(
    `${client.first_name} ${client.last_name}`
  );
  expect(getByTestId("client-address")).toHaveTextContent(client.address);
  expect(getByTestId("client-phone-number")).toHaveTextContent(
    client.phone_number
  );
});

it("contains a link to client page", () => {
  const { getByText } = render(
    <Router>
      <ClientCard client={client} />
    </Router>
  );

  expect(getByText("View Client")).toHaveAttribute(
    "href",
    `/partner/clients/${client.uuid}`
  );
});
