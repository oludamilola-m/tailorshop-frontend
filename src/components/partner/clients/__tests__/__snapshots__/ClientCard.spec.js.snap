// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="client"
>
  <div
    className="client__info"
  >
    <div
      className="client__icon"
    >
      <img
        alt="openq-b"
        src="http://odindesign-themes.com/vikinger/img/quest/openq-b.png"
      />
    </div>
    <p
      className="client__name"
      data-testid="client-name"
    >
       
      Olumuyiwa
       
      Margaret
    </p>
    <div
      className="client__details"
    >
      <div
        className="client__address"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map-marker-alt fa-w-12 "
          data-icon="map-marker-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          data-testid="client-address"
        >
          Gilder strasse 109, 36392 Berlin
        </span>
      </div>
      <p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-phone-alt fa-w-16 "
          data-icon="phone-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497.39 361.8l-112-48a24 24 0 0 0-28 6.9l-49.6 60.6A370.66 370.66 0 0 1 130.6 204.11l60.6-49.6a23.94 23.94 0 0 0 6.9-28l-48-112A24.16 24.16 0 0 0 122.6.61l-104 24A24 24 0 0 0 0 48c0 256.5 207.9 464 464 464a24 24 0 0 0 23.4-18.6l24-104a24.29 24.29 0 0 0-14.01-27.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          data-testid="client-phone-number"
          style={
            Object {
              "letterSpacing": "1px",
            }
          }
        >
          0801111111
        </span>
      </p>
    </div>
    <div
      className="client__jobs"
    >
      <p>
        5/7 jobs completed
      </p>
    </div>
    <div
      className="client__meta"
    >
      <div
        className="text-center"
      >
        <a
          className="btn btn--sm btn--link"
          href="/partner/clients/uuid"
          onClick={[Function]}
        >
          View Client
        </a>
      </div>
      <div
        className="flex"
      >
        <div
          className="edit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-edit fa-w-18 "
            data-icon="edit"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           Edit
        </div>
      </div>
    </div>
  </div>
</div>
`;
